### PR TITLE
feat(india): E Invoice Cancel Eway Bill is enabled.

### LINKED ISSUE #30887: feat(india): E Invoice Cancel Eway Bill is now available.
Adaequare has enabled their API for E-Way Bill Cancelation. Some Changes are required to enable the functionality in erpnext.

### PR DESCRIPTION
**Pull request by maharshivpatel**
Adaequare has enabled E-Way Bill Cancellation on thier API so necessary changes are made to enable functionality for the same.

This closes #30887 

`no-docs`

### CODE REVIEW

File: erpnext/regional/india/e_invoice/utils.py [hunk 1]
```diff
@@ -1300,10 +1299,9 @@ def generate_eway_bill(doctype, docname, **kwargs):
...
 @frappe.whitelist()
-def cancel_eway_bill(doctype, docname):
-	# TODO: uncomment when eway_bill api from Adequare is enabled
-	# gsp_connector = GSPConnector(doctype, docname)
-	# gsp_connector.cancel_eway_bill(eway_bill, reason, remark)
+def cancel_eway_bill(doctype, docname, eway_bill, reason, remark):
+	gsp_connector = GSPConnector(doctype, docname)
+	gsp_connector.cancel_eway_bill(eway_bill, reason, remark)
 
 	frappe.db.set_value(doctype, docname, "ewaybill", "")
```
Comment (nextchamp-saqib) [inline erpnext/regional/india/e_invoice/utils.py]: These 2 set_value are unnecessary now, these will be updated after the API request is successful

Comment (nextchamp-saqib) [inline erpnext/regional/india/e_invoice/utils.py:1306]: These 2 set_value are unnecessary now, these will be updated after the API request is successful

Comment (maharshivpatel) [reply erpnext/regional/india/e_invoice/utils.py]: done

**Review by nextchamp-saqib — DISMISSED** [review]

File: erpnext/regional/india/e_invoice/einvoice.js [hunk 2]
```diff
@@ -136,27 +136,58 @@ erpnext.setup_einvoice_actions = (doctype) => {
...
+									frappe.show_alert({
+										message: __('E-Way Bill Cancelled successfully'),
+										indicator: 'green'
+									}, 7);
+									frm.reload_doc();
+									d.hide();
+								},
+								error: () => {
+									frappe.show_alert({
+										message: __('E-Way Bill was not Cancelled '),
```
Comment (nextchamp-saqib) [inline erpnext/regional/india/e_invoice/einvoice.js]:
```suggestion
										message: __('E-Way Bill was not Cancelled'),
```

**Review by nextchamp-saqib — APPROVED** [review]

**Pull request merged**